zebra/kernel_netlink.c

```diff
@@ -970,23 +975,25 @@ void kernel_init(struct zebra_ns *zns)
 #endif
 
 	/* Register kernel socket. */
-	if (zns->netlink.sock > 0) {
-		/* Only want non-blocking on the netlink event socket */
-		if (fcntl(zns->netlink.sock, F_SETFL, O_NONBLOCK) < 0)
-			zlog_err("Can't set %s socket flags: %s",
-				 zns->netlink.name, safe_strerror(errno));
-
-		/* Set receive buffer size if it's set from command line */
-		if (nl_rcvbufsize)
-			netlink_recvbuf(&zns->netlink, nl_rcvbufsize);
-
-		netlink_install_filter(zns->netlink.sock,
-				       zns->netlink_cmd.snl.nl_pid);
-		zns->t_netlink = NULL;
-
-		thread_add_read(zebrad.master, kernel_read, zns,
-				zns->netlink.sock, &zns->t_netlink);
-	}
+	/* Only want non-blocking on the netlink event socket */
```

Comment:
This comment is outdated now that both sockets are non-blocking :) Please fix this and I should merge this PR.